DAH-11 chore(backend): Setup Express

jaysabe opened the pull request:
## Pull Request
 Added Express.js to the backend folder and updated .gitignore file

### Description

Completed Express.js set up in server directory. Then added json. 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)\
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### Issue ticket number and link

[DAH-11](https://igit.atlassian.net/jira/software/projects/DAH/boards/4?selectedIssue=DAH-11)

### Checklist

- [x] Self-reviewed my code prior to submission.
- [ ] Tests have been added or updated (if applicable).
- [ ] Documentation has been updated (if applicable).
- [ ] Screenshots added (if applicable).
- [x] This PR is ready for review.

### Screenshots


[DAH-11]: https://igit.atlassian.net/browse/DAH-11?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jaysabe committed:
Added the node_modules folder / package-lock.json
 file to the .gitignore file, then added Exoress.js to the backend folder

jaysabe committed:
Corrected .gitignore file for backend.

RickyC0626 reviewed (changes requested):
**thought:** I think you meant to put `package.json` in `backend` directory?
**suggestion:** Remove `node_modules` and move `package.json` and `package-lock.json` to `backend` directory. Perhaps we can also add another `.gitignore` file in root directory in case.

jaysabe committed:
Corrected node_modules and moved json packages to backend folder

RickyC0626 commented on backend/server/server.js:
```diff
@@ -0,0 +1,15 @@
+const app = express();
+const port = 3000;
+
+test = app.get("/", (req, res) => {
```
**suggestion:** Remove `test =` assignment, `app.get` will work without it.

RickyC0626 commented on backend/app.js:
```diff
@@ -0,0 +1 @@
+"use strict";
```
**thought:** This file is empty, maybe `server.js` could be moved here instead?

jaysabe replied:
Are you saying to bring server.js out to just backend/ ?

RickyC0626 replied:
Yeah, the contents of `server.js` can go here.

RickyC0626 commented on backend/package.json:
```diff
@@ -0,0 +1,32 @@
+  "version": "1.0.0",
+  "description": "parking app",
+  "main": "index.js",
+  "scripts": {
+    "test": "test_run"
+  },
```
**suggestion:** A `start` or `dev` script to run the app would be nice.

`dev` script could use something like [`nodemon`](https://nodemon.io) for hot reload.

`start` script could just run `node app.js`.

jaysabe replied:
added both for app.js

jaysabe committed:
Cleaned up the backend folders and files

jaysabe committed:
Added universal .gitignore to Dahlia project repo, and updated the
backend .gitignore to include the node_modules folder. Finally renamed the backend-notes.md file to README.md
to make it easier to find and read. Added Nodemon to dependencies.

jaysabe committed:
removed package-lock from .gitignore in both root
and backend.

jaysabe committed:
Slight edits to the .gitignore files to improve
efficiency.

CeeJayyy007 committed:
Merge branch 'dev' into DAH-11-Setup-node.js-express-in-backend

RickyC0626 commented on backend/.gitignore:
```diff
@@ -0,0 +1,28 @@
+
+# Miscellaneous
+*.log
+*.env
```
**suggestion:** 
```diff
*.env
+ !.env.example
```
This will be useful later for setting up env vars

RickyC0626 commented on backend/.gitignore:
```diff
@@ -0,0 +1,28 @@
+logs/
+
+# dotenv environment variables file
+.env
```
**thought:** Oh, we have a duplicate. Feel free to move `*.env` up here.

RickyC0626 commented on .gitignore:
```diff
@@ -0,0 +1,29 @@
+
+# Miscellaneous
+*.log
+*.env
```
**suggestion:** Same changes as `backend/.gitignore`

jaysabe replied:
Are you saying change in frontend .gitignore to match ?

jaysabe committed:
Edited .env behavior plan in backend/.gitignore and
 frontend/.gitignore.

jaysabe committed:
Merge branch 'DAH-11-Setup-node.js-express-in-backend' of https://github.com/Project-Dahlia/Dahlia into DAH-11-Setup-node.js-express-in-backend

jaysabe committed:
Edited root .gitignore to match BE and FE .gitignore
